test/valid-data/type-union/main.ts

```diff
@@ -6,6 +6,9 @@ type MyType4 = "s" | 1;
 type MyType5 = "s" | 1[];
 type MyType6 = ("s" | 1)[];
 
+type MyType7 = string | true;
+type MyType8 = "s" | true;
```

Comment:
feel free to bikeshed _a bit_ on naming this – I just blindly added the new items with sequential numbers so that I don’t change unrelated stuff